### DISCUSSION TITLE
Goal tracking via API or Ability to track goals without javascript or External tracking

### DISCUSSION THREAD

**Discussion by gregdotca** [plausible/analytics#1148] [Anything Else]
I'm trying to track a goal from an external site, but since the external site uses curl to retrieve the URL that I'm trying to track on the main site, the tracking doesn't seem to work, I'm guessing due to the javascript not loading correctly.

An example:

WebsiteA - The website I'm tracking with Plausible
WebsiteB - A completely separate website that doesn't use Plausible but that I own and have full control over

So essentially, when someone loads WebsiteB/trigger-curl.php, it uses curl to retrieve WebsiteA/record-goal.php. Although WebsiteA/record-goal.php contains the full Plausible tracking code (I've even tried things like onload in the body tag), nothing gets recorded when this URL is retrieved via curl, and I'm looking for a way to be able to record an external goal like this.

For context, WebsiteB is a piece of software I wrote, and WebsiteA is the software's primary website. My goal is to have it so that when someone installs the software, the goal gets triggered on WebsiteA so that I can keep track of the number of installations under the umbrella of the main website.

Is anything like this currently possible with Plausible? I have it working with my current analytics software, which I'm trying to move away from in favour of Plausible, and this is one of the last pain points I need to figure out in order to be able to make the switch permanently.

**Comment by metmarkosaric** [comment]
thanks @chetcuti! not sure i understand your use case. what's the url of this site you want to track? i can take a look if our script is loading correctly. in general, if the script is there it should track

if these two sites are on two different domains, then you may be seeing this drawback of using Plausible on two different domains with the same snippet https://plausible.io/docs/plausible-script#can-i-use-the-same-snippet-across-my-domain-and-subdomain

**Comment by gregdotca** [comment]
Here's a more straightforward example.

This URL is tracked by Plausible: example.com/index.php

When I load this URL in a web browser everything tracks properly, since my browser is executing the javascript. However this URL doesn't get loaded in a web browser, it gets loaded via a curl call that comes from a completely separate web application (one that is not tracked by Plausible). So the issue is that since I'm not loading the URL directly in a browser the javascript isn't loading, and therefore none of the Plausible tracking works. I'm just trying to figure out if there are any alternate methods that I could use to track a goal from this URL when it gets called via curl. An API call, a backup invisible pixel for noscript tracking, etc.

**Reply by metmarkosaric** [reply]
Ok, thanks for the details! There's no official way to do this. I'm aware that some people send events to us via API (for instance from mobile apps) but that's more of a hack as we don't have any official documentation on this. You're welcome to play around and try to make it work for yourself too. We do not have any noscript pixel tracking option either.

**Reply by gregdotca** [reply]
Your events API is exactly what I was looking for! Thanks for implementing this!

**Comment by ro-savage** [comment]
@metmarkosaric - I would be interested in something similar.

I would like to track a goal metric - but I would like to assign in to a UTM code. Because of how plausible does tracking (private!) we lose the UTM code on the user when they continue through to another app and sign up and/or enter credit card details and fully convert.

I would like to send a goal metic to plausible, along with the UTM codes that were associated with user (which we keep track off in our database).

Something like `plausible('Signup', {props: { utm_source: 'discord', utm_campaign: 'christmas-offer' }})`

**Comment by metmarkosaric** [comment]
You can take a look at our events API now https://plausible.io/docs/events-api

**Reply by ro-savage** [reply]
Thanks Marko - however it doesn't appear to be able to take utm tags. Just `referrer` and `screen_width`.

Is including UTM tags in the events API something that plausible would be open to?

This would allow goal/conversion tracking that can be directly correlated to the original campaigns

**Reply by gregdotca** [reply]
> You can take a look at our events API now https://plausible.io/docs/events-api

I just wanted to thank you for implementing this events API! This is exactly what I was looking for, and I just finished successfully implementing it.

**Comment by tomgibb** [comment]
Hi - new to Plausible. I'm passing UTMs in my URL in the `url` of the events API payload. The event shows up, but the UTMs are not showing up in the Top Sources widget in the dashboard. Perhaps I'm doing something wrong. 

My CURL: 

```
curl -i -X POST https://plausible.io/api/event \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.121 Safari/537.36 OPR/71.0.3770.284' \
  -H 'X-Forwarded-For: 127.0.0.1' \
  -H 'Content-Type: application/json' \
  --data '{"name":"api_test","url":"https://fakeurl.com?utm_campaign=tomtest&utm_source=airplane&utm_medium=skywriting","domain":"fakeurl.com","referrer":"tomtest"}'

```

Thank you.

**Reply by metmarkosaric** [reply]
hi @tomgibb! one note is that UTM and other query parameters are counted only when they start a new session on your site. unless they start a new session we don't display them in the Top Sources. could that be it?

**Reply by tomgibb** [reply]
Hi! Makes sense. I'll request that we randomize the test requests a bit more so we see what's going down. Thank you for the reply.
